hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/initialization/SystemContractsCreator.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.mono.state.initialization;
+
+import static com.hedera.node.app.service.mono.context.properties.StaticPropertiesHolder.STATIC_PROPERTIES;
+import static com.hedera.node.app.service.mono.txns.contract.ContractCreateTransitionLogic.STANDIN_CONTRACT_ID_KEY;
+import static com.hedera.node.app.spi.config.PropertyNames.BOOTSTRAP_SYSTEM_ENTITY_EXPIRY;
+import static com.hedera.node.app.spi.config.PropertyNames.CONTRACTS_SYSTEM_CONTRACTS;
+
+import com.hedera.node.app.service.mono.config.HederaNumbers;
+import com.hedera.node.app.service.mono.context.annotations.CompositeProps;
+import com.hedera.node.app.service.mono.context.properties.PropertySource;
+import com.hedera.node.app.service.mono.ledger.accounts.HederaAccountCustomizer;
+import com.hedera.node.app.service.mono.ledger.backing.BackingStore;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.service.mono.store.contracts.EntityAccess;
+import com.hedera.node.app.service.mono.store.contracts.precompile.ExchangeRatePrecompiledContract;
+import com.hedera.node.app.service.mono.store.contracts.precompile.HTSPrecompiledContract;
+import com.hedera.node.app.service.mono.store.contracts.precompile.PrngSystemPrecompiledContract;
+import com.hedera.node.app.spi.config.PropertyNames;
+import com.hederahashgraph.api.proto.java.AccountID;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.apache.tuweni.bytes.Bytes;
+import org.bouncycastle.util.encoders.Hex;
+
+/**
+ * Encapsulates all the logic for the creation of the enabled system precompiled contracts.
+ *
+ * <p>As of release 0.37.0, there are 3 such contracts:
+ * <ul>
+ *     <li>{@link HTSPrecompiledContract}</li>
+ *     <li>{@link ExchangeRatePrecompiledContract}</li>
+ *     <li>{@link PrngSystemPrecompiledContract}</li>
+ * </ul>
+ */
+@Singleton
+public class SystemContractsCreator {
```

Comment:
This creates a lot of new functionality in the mono-service module that we want to remove in the long run. Can we move the new functionality to another module?